### PR TITLE
Create crafting recipes for roller blades and skates

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary

SUMMARY: Content "Crafting recipes for items 'roller blades' and 'roller skates'"

#### Purpose of change

Allows crafting of roller blades and skates in case of skating emergency, plus recovery of craftable items to make one from the other

#### Describe the solution

Crafting recipes for roller skates and blades, based on several recipes with similar materials and complexity, especially hard hat for the plastic molding, which is the most complicated piece here. Uncrafting, new skate wheels, and item spawn for the skate wheels

#### Describe alternatives you've considered

No recipes, more recipes, skateboard, ice skates

#### Testing

Game loads with no errors, can build the things with the necessary skills, tools, and materials as expected.  Uncraft works as expected.
Screen Shot 2020-05-11 at 9 53 24 PM

### CODE REVIEW

**Comment by LyleSY** [comment]
Inspired by this reddit discussion https://www.reddit.com/r/cataclysmdda/comments/gfs3cj/how_the_hell_does_vormithrax_just_casually_walk/

**Comment by anothersimulacrum** [comment]
image
 These don't look like they should give normal boots.

**Comment by LyleSY** [comment]
Agreed, I was thinking it would make sense to get damaged boots out of them, but I don't see a way to do that

**Comment by LyleSY** [comment]
...and now I see that the boots in game are all made of leather, not very common for roller blades. Uncrafting might not make a ton of sense here after all.

**Comment by LyleSY** [comment]
I think I've found and murdered all the bugs now, that Travis error looks like it's about nested containers

File: data/json/recipes/armor/feet.json [hunk 1]
```diff
@@ -555,5 +555,40 @@
...
+  {
+    "result": "rollerskates",
+    "type": "recipe",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_FEET",
+    "skill_used": "fabrication",
+    "difficulty": 5,
+    "time": "60 m",
+    "book_learn": [ [ "textbook_tailor", 5 ], [ "textbook_fabrication", 5 ] ],
+    "using": [ [ "welding_standard", 6 ] ],
```
Comment (kevingranade) [inline data/json/recipes/armor/feet.json:590]: Not welding, forging.

File: data/json/recipes/armor/feet.json [hunk 2]
```diff
@@ -555,5 +555,40 @@
...
+    "skill_used": "fabrication",
+    "skills_required": [ "tailor", 2 ],
+    "difficulty": 5,
+    "time": "90 m",
+    "book_learn": [ [ "textbook_tailor", 5 ], [ "textbook_fabrication", 5 ] ],
+    "using": [ [ "welding_standard", 6 ] ],
+    "tools": [ [ [ "mold_plastic", -1 ] ], [ [ "surface_heat", 15, "LIST" ] ] ],
+    "qualities": [ { "id": "HAMMER", "level": 1 }, { "id": "DRILL", "level": 1 }, { "id": "SAW_M", "level": 1 } ],
+    "components": [
+      [ [ "xlframe", 1 ] ],
```
Comment (kevingranade) [inline data/json/recipes/armor/feet.json:573]: This is like bicycle tubes???

File: data/json/recipes/armor/feet.json [hunk 3]
```diff
@@ -555,5 +555,40 @@
...
+    "result": "roller_blades",
+    "type": "recipe",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_FEET",
+    "skill_used": "fabrication",
+    "skills_required": [ "tailor", 2 ],
+    "difficulty": 5,
+    "time": "90 m",
+    "book_learn": [ [ "textbook_tailor", 5 ], [ "textbook_fabrication", 5 ] ],
+    "using": [ [ "welding_standard", 6 ] ],
```
Comment (kevingranade) [inline data/json/recipes/armor/feet.json:569]: Also forging, not welding.

File: data/json/uncraft/armor/boots.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+[
+  {
+    "result": "roller_blades",
+    "type": "uncraft",
+    "skill_used": "fabrication",
+    "difficulty": 2,
+    "time": "600 s",
+    "qualities": [ { "id": "PRY", "level": 1 } ],
+    "components": [
+      [ [ "material_aluminium_ingot", 10 ] ],
```
Comment (kevingranade) [inline data/json/uncraft/armor/boots.json:10]: Where are you getting ingots?

**Comment by LyleSY** [comment]
> These are still way too approximate, especially with as OP as the game makes these things.

Can you give me an example of what you have in mind here? I'm looking at end game weapons like katana and zweihander and this is much richer.  Katana example:
`    "type": "recipe",
    "result": "katana",
    "category": "CC_WEAPON",
    "subcategory": "CSC_WEAPON_CUTTING",
    "skill_used": "fabrication",
    "difficulty": 9,
    "time": "8 h",
    "book_learn": [ [ "textbook_weapeast", 8 ] ],
    "using": [ [ "blacksmithing_standard", 8 ], [ "steel_standard", 2 ] ],
    "qualities": [ { "id": "CHISEL", "level": 3 } ],
    "tools": [ [ [ "swage", -1 ] ] ],
    "components": [ [ [ "2x4", 2 ], [ "stick", 4 ] ], [ [ "fur", 2 ], [ "leather", 2 ] ] ]`

**Pull request merged**